File: BankKycSystemdotnet/BankKycSystem/Controllers/AuthController.cs
using BankKycSystem.DTOs;
using BankKycSystem.Services.Interfaces;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;

namespace BankKycSystem.Controllers
{
    [ApiController]
    [Route("api/[controller]")]
    public class AuthController : ControllerBase
    {
        private readonly IAuthService _authService;
        private readonly IOtpService _otpService;

        public AuthController(IAuthService authService, IOtpService otpService)
        {
            _authService = authService;
            _otpService = otpService;
        }

        // Step 1: Send OTP
        [HttpPost("login")]
        public async Task<IActionResult> Login([FromBody] LoginRequestDTO dto)
        {
            var result = await _authService.InitiateLoginAsync(dto);
            return Ok(result);
        }

        // Step 2: Confirm OTP and get JWT
        [HttpPost("confirm-login")]
        public async Task<IActionResult> ConfirmLogin([FromBody] ConfirmLoginRequestDTO dto)
        {
            var response = await _authService.ConfirmLoginAsync(dto);
            return Ok(response);
        }

        // Resend OTP
        [HttpPost("resend-otp")]
        public async Task<IActionResult> ResendOtp([FromBody] EmailDTO dto)
        {
            await _otpService.ResendOtpAsync(dto.Email);
            return Ok($"OTP resent to email: {dto.Email}");
        }

        // Optional direct OTP verify
        [HttpPost("verify-otp")]
        public async Task<IActionResult> VerifyOtp([FromQuery] string email, [FromQuery] string otp)
        {
            bool valid = await _otpService.ValidateOtpAsync(email, otp);
            return valid ? Ok("OTP verified") : BadRequest("Invalid or expired OTP");
        }
    }
}

File: bank-kyc-system/src/main/java/com/bank/project/controller/AuthController.java
package com.bank.project.controller;

import com.bank.project.dto.*;
import com.bank.project.service.AuthService;
import com.bank.project.service.OtpService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import jakarta.validation.Valid;

@RestController
@RequestMapping("/api/auth")
@RequiredArgsConstructor
@CrossOrigin
public class AuthController {

    private final AuthService authService;
    private final OtpService otpService;

    // Step 1: Send OTP to user (Admin/Clerk/Manager)
    @PostMapping("/login")
    public ResponseEntity<?> login(@Valid @RequestBody LoginRequestDTO dto) {
        try {
            String result = authService.initiateLogin(dto);
            if (result == null || result.trim().isEmpty()) {
                return ResponseEntity.status(HttpStatus.UNAUTHORIZED)
                        .body("Invalid email/phone or password.");
            }
            return ResponseEntity.ok(result);
        } catch (Exception e) {
            return ResponseEntity.status(HttpStatus.UNAUTHORIZED)
                    .body("Invalid email/phone or password.");
        }
    }

    // Step 2: Confirm OTP and generate JWT token
    @PostMapping("/confirm-login")
    public ResponseEntity<?> confirmLogin(@RequestBody ConfirmLoginRequestDTO dto) {
        try {
            LoginResponseDTO response = authService.confirmLogin(dto);
            if (response == null || response.getToken() == null) {
                return ResponseEntity.status(HttpStatus.UNAUTHORIZED)
                        .body("Invalid or expired OTP.");
            }
            return ResponseEntity.ok(response);
        } catch (Exception e) {
            return ResponseEntity.status(HttpStatus.UNAUTHORIZED)
                    .body("Invalid or expired OTP.");
        }
    }

    // Optional: Resend OTP if needed
    @PostMapping("/resend-otp")
    public ResponseEntity<String> resendOtp(@RequestBody EmailDTO request) {
        otpService.resendOtp(request.getEmail());
        return ResponseEntity.ok("OTP resent to email: " + request.getEmail());
    }

    // Optional: Direct OTP validation API (if needed for custom flows)
    @PostMapping("/verify-otp")
    public ResponseEntity<String> verifyOtp(@RequestParam String email, @RequestParam String otp) {
        boolean valid = otpService.validateOtp(email, otp);
        return valid
                ? ResponseEntity.ok("OTP verified")
                : ResponseEntity.badRequest().body("Invalid or expired OTP");
    }
}
